x-pack/plugins/security/server/authentication/authenticator.ts

```diff
@@ -637,7 +637,7 @@ export class Authenticator {
     //  4. Request isn't attributed with HTTP Authorization header
     return (
       canRedirectRequest(request) &&
-      !sessionValue &&
+      (!sessionValue || !sessionValue.username) &&
```

Comment:
nit: what do you think about introducing a `isSessionAuthenticated` function (or similar) for this? I know it's a very basic check, but since this will be the second place for such a check (https://github.com/elastic/kibana/blob/899f8e1e9967225f638914cea920aa3aadaa6b1c/x-pack/plugins/security/server/authentication/authenticator.ts#L561), it might make sense to consolidate the logic, if nothing else for readability.

Reply:
Yeah, I think it makes sense, will do, thanks!

Reply:
While fixing this I realized that I cannot backport it to 7.9.x since we started to store username in the session for all providers only in 7.10 (with server-side sessions) that means we don't have an easy way to distinguish authenticated session from unauthenticated in 7.9 :cry: